src/Compilers/CSharp/Test/Semantic/Semantics/DynamicTests.cs

```diff
@@ -3550,5 +3550,260 @@ .maxstack  8
+
+            compilation2.VerifyDiagnostics(
+                // (10,18): error CS0120: An object reference is required for the non-static field, method, or property '_Worksheet.Range[object, object]'
+                //         var z1 = Worksheet.Range[x, y];
+                Diagnostic(ErrorCode.ERR_ObjectRequired, "Worksheet.Range[x, y]").WithArguments("Microsoft.Office.Interop.Excel._Worksheet.Range[object, object]").WithLocation(10, 18)
+                );
+        }
+
+        [ClrOnlyFact(ClrOnlyReason.Ilasm)]
+        [WorkItem(204561, "https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems?id=204561&_a=edit")]
+        public void Bug204561_02()
+        {
+            var iLSource = @"
+.class public auto ansi WithIndexer
+       extends [mscorlib]System.Object
+{
+  .method public specialname rtspecialname 
+          instance void  .ctor() cil managed
+  {
+    // Code size       7 (0x7)
+    .maxstack  8
+    IL_0000:  ldarg.0
+    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
+    IL_0006:  ret
+  } // end of method WithIndexer::.ctor
+
+  .method public specialname static object 
+          get_Indexer(object x,
+                      object y) cil managed
+  {
+    // Code size       18 (0x12)
+    .maxstack  1
+    .locals init (object V_0)
+    IL_0000:  nop
+    IL_0001:  ldstr      ""Indexer""
+    IL_0006:  call       void [mscorlib]System.Console::WriteLine(string)
+    IL_000b:  nop
+    IL_000c:  ldnull
+    IL_000d:  stloc.0
+    IL_000e:  br.s       IL_0010
+
+    IL_0010:  ldloc.0
+    IL_0011:  ret
+  } // end of method WithIndexer::get_Indexer
+
+  .method public specialname static void 
+          set_Indexer(object x,
+                      object y,
+                      object 'value') cil managed
+  {
+    // Code size       2 (0x2)
+    .maxstack  8
+    IL_0000:  nop
+    IL_0001:  ret
+  } // end of method WithIndexer::set_Indexer
+
+  .method public static object  MIndexer(object x,
+                                         object y) cil managed
+  {
+    // Code size       18 (0x12)
+    .maxstack  1
+    .locals init (object V_0)
+    IL_0000:  nop
+    IL_0001:  ldstr      ""MIndexer""
+    IL_0006:  call       void [mscorlib]System.Console::WriteLine(string)
+    IL_000b:  nop
+    IL_000c:  ldnull
+    IL_000d:  stloc.0
+    IL_000e:  br.s       IL_0010
+
+    IL_0010:  ldloc.0
+    IL_0011:  ret
+  } // end of method WithIndexer::MIndexer
+
+  .property object Indexer(object,
+                           object)
+  {
+    .get object WithIndexer::get_Indexer(object,
+                                         object)
+    .set void WithIndexer::set_Indexer(object,
+                                       object,
+                                       object)
+  } // end of property WithIndexer::Indexer
+} // end of class WithIndexer
+";
+
+            MetadataReference reference = CompileIL(iLSource, appendDefaultHeader: true, embedInteropTypes: false);
```

Comment:
Can an assembly of this structure only be created in IL? Just asking because if it could also be created in VB it may be interesting to have a test with VB code to ensure reasonable interop.